src/redux/oauth/oauthSlice.ts

```diff
@@ -0,0 +1,30 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit"
+import type { RootState } from "../store"
+
+type sliceState = { isSignedIn: boolean; userId: null | string }
+
+export const initialState: sliceState = {
+  userId: null,
+  isSignedIn: false,
+}
+export const oauthSlice = createSlice({
+  name: "oauth",
+  initialState,
+  reducers: {
+    signIn: (state, action: PayloadAction<string>) => {
+      state.isSignedIn = true
+      state.userId = action.payload
+    },
+    signOut: (state) => {
+      state.isSignedIn = false
+      state.userId = "null"
```

Comment:
Should this be `null` instead of `"null"`?

Reply:
Yes! I forgot to change that when I made the type of sliceState -- I forgot that I put that there as a placeholder until I figured out how to have it actually null.